src/pages/Order/index.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Box,
+  Button,
+  ChakraProvider,
+  Checkbox,
+  Flex,
+  HStack,
+  Image,
+  Input,
+  Select,
+  Text,
+  Textarea,
+  VStack,
+} from '@chakra-ui/react';
+import type { FieldError } from 'react-hook-form';
+import { useForm } from 'react-hook-form'; // react-hook-form import 추가
+import { useLocation } from 'react-router-dom';
+
+export const Order = () => {
+  const location = useLocation();
+  const { productDetail, quantity, price } = location.state || {};
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm(); // useForm 훅을 사용하여 폼 상태와 메소드 초기화
+
+  // ref 대신 register 함수 사용
+  const messageRef = register('message', {
+    required: '카드 메시지를 입력해주세요.',
+    maxLength: { value: 100, message: '카드 메시지는 100자 이내로 입력해주세요.' },
+  });
+  const receiptNumberRef = register('receiptNumber', {
+    required: '현금영수증 번호를 입력해주세요.',
```

Comment:
> 질문 : [카드 메세지를 작성하고 현금 영수증을 체크하지 않은 경우]와 [카드메세지를 작성하고 현금영수증을 체크하고 휴대폰 번호를 적은 경우] 위 두 상태에 대해 주문이 접수되었다는 alert()를 띄우려고 onSubmit() 내부에 조건문을 작성하였는데, 처음에 1번 경우로 제출한 경우에는 제대로 제출되지 않는 것 같습니다. (주문시 콘솔이 찍히지 않고, alert()도 안 뜸) 2번으로 한 번 제출한 후에는 1번 경우에도 alert()가 뜨는데, 처음에 1번 경우가 alert()가 뜨지 않는 이유가 궁금합니다..! validation 자체는 잘 되는것으로 확인됩니다.

`receiptNumber`가 항상 필요하도록 required가 적용되어 있는게 원인입니다. 조건부로 현금영수증 신청에 체크되어 있을 때 필수값으로 유효성 검사가 되는게 맞으니까요. `const receiptNumber = watch('receiptNumber');`를 28번 라인 밑에 선언하고 아래처럼 변경이 필요해보여요.


```suggestion
    required: receiptNumber ? '현금영수증 번호를 입력해주세요.' : false,
```